ql/experimental/swaptions/haganirregularswaptionengine.cpp

```diff
@@ -334,48 +313,50 @@ namespace QuantLib {
         s1d.setMaxEvaluations(10000);
         s1d.setLowerBound(minLambda);
         s1d.setUpperBound(maxLambda);
-        s1d.solve(basket,1.0e-8,0.01, minLambda, maxLambda);
+        s1d.solve(basket, 1.0e-8, 0.01, minLambda, maxLambda);
 
 
         /////////////////////////////////////////////////////////////////////////////////////////////////
-        //  compute the price of the irreg swaption as the sum of the prices of the regular swaptions  //
+        //  compute the price of the irreg swaption as the sum of the prices of the regular
+        //  swaptions  //
         /////////////////////////////////////////////////////////////////////////////////////////////////
 
 
-        results_.value = HKPrice(basket,exercise_);
+        results_.value = HKPrice(basket, exercise_);
     }
 
 
-
     /////////////////////////////////////////////////////////////////////////////////////////
     // Computes irregular swaption price according to P.J. Hunt, J.E. Kennedy:             //
     // "Implied interest rate pricing models", Finance Stochast. 2, 275-293 (1998)         //
     /////////////////////////////////////////////////////////////////////////////////////////
 
-    Real  HaganIrregularSwaptionEngine::HKPrice(Basket& basket,ext::shared_ptr<Exercise>& exercise) const {
+    Real HaganIrregularSwaptionEngine::HKPrice(Basket& basket,
+                                               ext::shared_ptr<Exercise>& exercise) const {
+
+        // Swaption Engine: assumes that the swaptions exercise date equals the swap start date
+        QL_REQUIRE((volatilityStructure_->volatilityType() == Normal),
+                   "swaptionEngine: only normal volatility implemented.");
+
 
-        // Black 76 Swaption Engine: assumes that the swaptions exercise date equals the swap start date
-        ext::shared_ptr<PricingEngine> blackSwaptionEngine = 
-             ext::shared_ptr<PricingEngine>(new BlackSwaptionEngine(termStructure_,volatilityStructure_));
+        ext::shared_ptr<PricingEngine> swaptionEngine = ext::shared_ptr<PricingEngine>(
+            new BachelierSwaptionEngine(termStructure_, volatilityStructure_));
```

Comment:
I'm guessing Hunt and Kennedy used Bachelier and we were using Black instead because we couldn't use normal vols yet—is that correct?